From dc00dbb695d5e9d748fa633d1009f21329ecd84f Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Thu, 31 Mar 2016 00:51:46 +0530
Subject: [PATCH 18/19] SQUASHME! [VPG]: drm/i915: Avoid clearing test request
 till handling it

Test requests and reply need to be handled at a single point to
avoid racing issues. This patch avoids clearing of test request
in intel_dp_detect that was missed in the earlier patch.

This patch has to be squashed into
Change-Id: I8458456f305acf5bc6771ab6eec38972fc4d2b95
Author: Thulasimani,Sivakumar <sivakumar.thulasimani@intel.com>
Date:   Tue Sep 1 15:33:00 2015 +0530
FOR_UPSTREAM [VPG]: drm/i915: Cleanup panel IRQ handling

Change-Id: Ic6487c915d8439175d4ec9821c8ce76b7c061a19
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498193
---
 drivers/gpu/drm/i915/intel_dp.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 96d6258..54f549e 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4902,15 +4902,18 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 	/* Try to read the source of the interrupt */
 	if (intel_dp->dpcd[DP_DPCD_REV] >= 0x11 &&
 	    intel_dp_get_sink_irq(intel_dp, &sink_irq_vector)) {
+		if (sink_irq_vector & DP_AUTOMATED_TEST_REQUEST) {
+			intel_dp_handle_test_request(intel_dp, false);
+			sink_irq_vector &= ~DP_AUTOMATED_TEST_REQUEST;
+		}
+		if (sink_irq_vector & (DP_CP_IRQ | DP_SINK_SPECIFIC_IRQ))
+			DRM_DEBUG_DRIVER("CP or sink specific irq unhandled\n");
+
 		/* Clear interrupt source */
 		drm_dp_dpcd_writeb(&intel_dp->aux,
 				   DP_DEVICE_SERVICE_IRQ_VECTOR,
 				   sink_irq_vector);
 
-		if (sink_irq_vector & DP_AUTOMATED_TEST_REQUEST)
-			intel_dp_handle_test_request(intel_dp, false);
-		if (sink_irq_vector & (DP_CP_IRQ | DP_SINK_SPECIFIC_IRQ))
-			DRM_DEBUG_DRIVER("CP or sink specific irq unhandled\n");
 	}
 
 	/* if simulation was in progress clear the flag & skip upfront */
-- 
1.9.1

